protocols/relay/src/behaviour.rs

```diff
@@ -184,7 +184,10 @@ pub enum Event {
         error: inbound_hop::Error,
     },
     /// An inbound reservation request has been denied.
-    ReservationReqDenied { src_peer_id: PeerId },
+    ReservationReqDenied {
+        src_peer_id: PeerId,
+        status: proto::Status,
```

Comment:
If we are to introduce this, we should create a enum `Status` that we can map `proto::Status` to. Thoughts?